### PR TITLE
Document plugins

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
Initial documentation for the new plugin system

https://deploy-preview-881--porter.netlify.com/mixins/
https://deploy-preview-881--porter.netlify.com/plugins/
https://deploy-preview-881--porter.netlify.com/mixins-vs-plugins/
https://deploy-preview-881--porter.netlify.com/plugins/azure/

# What issue does it fix
The lack of documentation 😎

# Notes for the reviewer
* Is there anything else we want to document while we are feeling spiffy?
* The azure page says to install with `porter plugins install` which hasn't been merged yet. Depending on timing, I can update the page to use the instructions on the readme, which have them build and install manually.

# Checklist
- [ ] Unit Tests
- [x] Documentation
  - [ ] Documentation Not Impacted

### CODE REVIEW

File: docs/content/mixins-vs-plugins.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+title: Mixins vs. Plugins
+description: What is the difference between a Porter mixin and a plugin? When would you use one instead of another?
+---
+
+[Mixins](/mixins/) are the building blocks for authoring bundles. They
+understand both Porter and CNAB, so you don't have to, and ideally have built in
+extra logic to make your bundles easier to author and more robust.  They
```
Comment (vdice) [inline docs/content/mixins-vs-plugins.md]: nit: consider hyphenating for `built-in`

Comment (vdice) [reply docs/content/mixins-vs-plugins.md]: nit: consider reducing adjectives in `built in extra logic`

File: docs/content/mixins-vs-plugins.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+involved when the bundle is built, assisting with generating the Dockerfile for
```
Comment (vdice) [inline docs/content/mixins-vs-plugins.md]: Perhaps `They involved` -> `They are involved`?

File: docs/content/mixins-vs-plugins.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+actions in your porter.yaml when your bundle is run.
+
+I like to think of mixins as paint colors on your paint palette 🎨 that you use
+to **create** your bundle. Like paints, mixins are **composable** and let you
+build something new, more than the sum of its parts, limited only by your
+imagination.
+
+[Plugins](/plugins/) **extend** the Porter client itself, **reimplementing**
+Porter's default functionality. There are fixed extension points in Porter with
+a defined interface. For example by default Porter saves claims and credential
```
Comment (vdice) [inline docs/content/mixins-vs-plugins.md]: nit: consider `For example, by default, Porter saves...` or similar

File: docs/content/mixins-vs-plugins.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+sets on the local filesystem to ~/.porter. A plugin can change that behavior to
+save them to cloud storage instead.
+
+What both mixins and plugins have in common is that anyone can create their own
+and distribute them, just like the ones that we install with Porter by default.
+The Porter team is committed to making our plugin and mixin ecosystem a level
+playing field with a low barrier to entry. Check our out [Mixin Developer
```
Comment (vdice) [inline docs/content/mixins-vs-plugins.md]: `Check our out` -> `Check out our`

File: docs/content/mixins/_index.md [hunk 5]
```diff
@@ -1,15 +1,7 @@
...
-Mixins make Porter special, find them, use them, [create your own](/mixin-dev-guide).
-
-* [exec](/mixins/exec)
-* [helm](/mixins/helm)
-* [kubernetes](/mixins/kubernetes)
-* [az](/mixins/az)
-* [aws](/mixins/aws)
-* [gcloud](/mixins/gcloud)
-* [terraform](/mixins/terraform)
+Mixins make Porter special, they are the building blocks that you use when
```
Comment (vdice) [inline docs/content/mixins/_index.md]: nit: consider adjusting punctuation.  Perhaps `Mixins make Porter special.  They are the...` ?  Or, a semicolon in place of the comma.

File: docs/content/plugins/azure.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+```
+porter plugin install azure
+```
+
+## Plugin Configuration
+
+### Storage
+
+Storage plugins allow Porter to store data, such as claims and credentials, in the Azure's cloud.
```
Comment (vdice) [inline docs/content/plugins/azure.md]: Perhaps `..in Azure's cloud` or `..in the Azure cloud` instead?

File: docs/content/mixins-vs-plugins.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+in your porter.yaml when your bundle is run.
+
+I like to think of mixins as paint colors on your paint palette 🎨 that you use
+to **create** your bundle. Like paints, mixins are **composable** and let you
+build something new, more than the sum of its parts, limited only by your
+imagination.
+
+[Plugins](/plugins/) **extend** the Porter client itself, **reimplementing**
+Porter's default functionality. There are fixed extension points in Porter with
+a defined interface. For example, Porter saves claims and credential sets local
```
Comment (vdice) [inline docs/content/mixins-vs-plugins.md]:
```suggestion
a defined interface. For example, Porter saves claims and credential sets using the local
```

**Comment by carolynvs** [comment]
Gonna hold off on merging this #771 is complete.

**Comment by dev-drprasad** [comment]
@carolynvs Looks great to me. Just one suggestion: maybe providing an example for mixins, just like for plugin you did, in mixins-vs-plugins page helps for beginners i think

**Comment by carolynvs** [comment]
* Added doc on plugin types: https://deploy-preview-881--porter.netlify.com/plugins/types/
* Added deeper links to the pages and cleaned up existing text about creds to make sure people know about plugins.

**Pull request merged**